scripts/lis_hyperv_platform/main.ps1

```diff
@@ -14,36 +12,88 @@ param(
-if ($LastExitCode -ne 0) {
-    throw
+function MyTest-Path {
+    param(
+        [String] $Path
+    )
+    if (!(Test-Path $Path)) {
+       throw "Path $Path not found"
+    }
+
 }
 
-$ip = Get-IP $InstanceName $VMCheckTimeout
-if ($ip) {
-    $throtleTimeStep = 5
-    $retryPeriod = 10
-    $retryTimes = 10
-    while ($retryTimes -gt 0) {
-        Write-Host "Trying to connect via SSH to $ip..."
-        & ssh.exe -tt -o StrictHostKeyChecking=no -i "$JobPath\$InstanceName-id-rsa" ubuntu@$ip
-        if ($LastExitCode) {
-            Write-Host "Failed to connect to $ip with error code: $LastExitCode"
-        }
-        $retryPeriod += $throtleTimeStep
-        $retryTimes = $retryTimes - 1
-        Start-Sleep $retryPeriod
+function Prepare-LocalEnv {
+    param(
+        [String] $SharedStoragePath,
+        [String] $JobId
+    )
+
+    $path = "/var/lib/lava/dispatcher/tmp/$JobId"
+    $remotePath = "H:\$JobId"
+    $localPath = "C:$path"
+
+    $SharedStoragePath = $SharedStoragePath.Replace("\\", "\")
+
+    net use H: $SharedStoragePath /persistent:NO 2>&1 | Out-Null
+    if ($LastExitCode) {
+        throw
     }
-} else {
-    throw "IP for instance $InstanceName not exposed."
+
+
+    MyTest-Path $remotePath
+
+    New-Item -Path $localPath -ItemType "directory" | Out-Null
+    Copy-Item -Path "$remotePath/*" -Destination $localPath -Force -Recurse
+
+    $localVHDPath = (Get-ChildItem -Filter "ubuntu-cloud.vhdx" -Path $localPath -Recurse ).FullName
+    MyTest-Path $localVHDPath
+
+    $lavaToolDisk = (Get-ChildItem -Filter "lava-guest.vhdx" -Path $localPath -Recurse ).FullName
+    MyTest-Path $lavaToolDisk
+
+    $remoteVHDPath = (Get-ChildItem -Filter "ubuntu-cloud.vhdx" -Path $remotePath -Recurse ).FullName
+    $remotePath = Split-Path -Parent $remoteVHDPath
+
+    return @($localVHDPath, $lavaToolDisk, $remotePath)
+}
+
+function Expand-URL {
+    param(
+        [String] $KernelUrl,
+        [String] $KernelVersion
+    )
+
+    $kernelURLExpanded = @()
+    $kernelURLExpanded += "{0}/hyperv-daemons_{1}_amd64.deb" -f @($KernelURL, $KernelVersion)
+    $kernelURLExpanded += "{0}/linux-headers-{1}_{1}-10.00.Custom_amd64.deb" -f @($KernelURL, $KernelVersion)
+    $kernelURLExpanded += "{0}/linux-image-{1}_{1}-10.00.Custom_amd64.deb" -f @($KernelURL, $KernelVersion)
+
+    return $kernelURLExpanded
 }
 
-& "$scriptPath\tear_down_env.ps1" $JobPath $InstanceName 
-if ($LastExitCode -ne 0) {
-    throw
+function Main {
+    $Error.Clear()
+
+    $return = Prepare-LocalEnv $SharedStoragePath $JobId
+    $localVHDPath = $return[0]
+    $lavaToolDisk = $return[1]
+    $remoteJobFolder = $return[2]
+
+    $expandedURL = Expand-URL $KernelURL $KernelVersion
+    $jobPath = Split-Path -Parent $localVHDPath
+    
+    & "$scriptPath\setup_env.ps1" $jobPath $localVHDPath $UserdataPath $expandedURL $InstanceName $MkIsoFS $lavaToolDisk
+    if ($Error.Count -ne 0) {
```

Comment:
ditto

Reply:
this is a replacement for $LASTEXITCODE but can catch other errors too